vertica_python/vertica/connection.py

```diff
@@ -501,11 +503,31 @@ def _socket(self):
             if load_balance_options:
                 raw_socket = self.balance_load(raw_socket)
 
-            # enable SSL
+            # enable TLS
+            tlsmode_options = self.options.get('tlsmode')
             ssl_options = self.options.get('ssl')
-            self._logger.debug('SSL option is {0}'.format('enabled' if ssl_options else 'disabled'))
-            if ssl_options:
-                raw_socket = self.enable_ssl(raw_socket, ssl_options)
+            # If TLSmode option and SSL option are set, TLSmode option takes precedence.
+            ssl_context = None
+            if tlsmode_options is not None:
```

Comment:
Would this logic be clearer as a function that takes tlsmode_options and ssl_options and returns the correct tlsmode?
`tlsmode = _initTLSMode(self.options.get('tlsmode'), self.options.get('ssl'))`